firmware/hw_layer/board_overrides.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * @file board_overrides.h
+ * @brief Board-specific override mechanism
+ *
+ * This header provides a centralized way to define and manage board-specific overrides.
+ * Custom boards can implement their own specialized behavior by overriding these function pointers.
+ *
+ * Usage Example:
+ *
+ * 1. Define your custom function:
+ *    void myBoardCustomHello() {
+ *        // Your custom implementation
+ *    }
+ *
+ * 2. Set up the override in your board_configuration.cpp:
+ *    void setup_custom_board_overrides() {
+ *        custom_board_boardSayHello = myBoardCustomHello;
+ *    }
+ *
+ *
+ * @date: jul 09, 2025
+ * @author FDSoftware
+ */
+
+#pragma once
+#include <functional>
+#include <optional>
+
+// function with no parameters and returning void
+using setup_custom_board_overrides_type = void (*)();
+
+/**
+ * @brief Pre-HAL initialization override point
+ * Allows boards to perform custom initialization before HAL is initialized
+ * Example: special pin setup or hardware initialization needed before HAL
+ */
+// extern std::optional<setup_custom_board_overrides_type> custom_board_preHalInit;
```

Comment:
let's go without example, just convert a few methods and have those act as examples

Reply:
> let's go without example, just convert a few methods and have those act as examples

done in 68aff94 for nucleo F429
and in c3a2436 for nucleo H743

which are the only ones that use the old `preHalInit`